wc/src/main.rs

```diff
@@ -0,0 +1,228 @@
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::{self, BufReader};
+use std::process;
+
+use clap::{load_yaml, App, ArgMatches};
+
+const PRINT_LINES: u8 = 0x1;
+const PRINT_WORDS: u8 = 0x2;
+const PRINT_CHARS: u8 = 0x4;
+const PRINT_BYTES: u8 = 0x8;
+const PRINT_MAX_LINE_LEN: u8 = 0x10;
+const DEFAULT_FLAGS: u8 = PRINT_LINES | PRINT_WORDS | PRINT_CHARS;
+
+fn main() {
+    let yaml = load_yaml!("wc.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let filenames: Vec<String> = {
+        if let Some(values) = matches.values_of("FILE") {
+            let mut v = Vec::new();
+            for value in values {
+                v.push(value.to_owned());
+            }
+            v
+        } else {
+            vec![String::from("-")]
+        }
+    };
+
+    let flags = parse_flags(&matches);
+
+    let mut total_result = WcResult::default();
+    for filename in &filenames {
+        let result = if filename == "-" {
+            let stdin = io::stdin();
+            wc(stdin.lock(), flags)
+        } else {
+            match File::open(filename) {
+                Ok(file) => wc(file, flags),
+                Err(err) => Err(err),
+            }
+        };
+
+        match result {
+            Err(err) => eprintln!("wc: {}: {}", filename, err),
+            Ok(result) => {
+                print_result(filename, &result);
+                total_result = total_result.combine(result);
+            }
+        }
+    }
+
+    if filenames.len() > 1 {
+        print_result("total", &total_result);
+    }
+
+    process::exit(0);
```

Comment:
`process::exit()` do not drop what needs to be dropped, so when it's finishing successfully it's a better pattern to just `return` in early ends, or in this case just let the main finish and drop what needs to be dropped, while if needs other exit codes, drop everything manually and `process::exit()`